assignments/lambda-classes.js

```diff
@@ -1 +1,148 @@
+  class Instructor extends Person {
+    constructor(props) {
+      super(props);
+      this.specialty = props.specialty;
+      this.favLanguage = props.favLanguage;
+      this.catchPhrase = props.catchPhrase;
+    }
+  
+    demo(subject) {
+      console.log(`Today we are learning about ${subject}.`);
+    }
+    grade(student, subject) {
+      console.log(`${student.name} receives a perfect score on ${subject}.`);
+    }
+  }
+  // #### Student
+  // * Now we need some students!
+  // * Student uses the same attributes that have been set up by Person
+  // * Student has the following unique props:
+  //   * `previousBackground` i.e. what the Student used to do before Lambda School
+  //   * `className` i.e. CS132
+  //   * `favSubjects`. i.e. an array of the student's favorite subjects ['Html', 'CSS', 'JavaScript']
+  // * Student has the following methods:
+  //   * `listsSubjects` a method that logs out all of the student's favoriteSubjects one by one.
+  //   * `PRAssignment` a method that receives a subject as an argument and logs out that the `student.name has submitted a PR for {subject}`
+  //   * `sprintChallenge` similar to PRAssignment but logs out `student.name has begun sprint challenge on {subject}`
+  class Student extends Person {
+    constructor(props) {
+      super(props);
+      this.previousBackground = props.previousBackground;
+      this.className = props.className;
+      this.favSubjects = props.favSubjects;
+    }
+  
+    listsSubjects() {
+      this.favSubjects.forEach(subject => {
+        console.log(subject);
+      });
+    }
+    PRAssignment(subject) {
+      console.log(`${this.name} has submitted a PR for ${subject}.`);
+    }
+    sprintChallenge(subject) {
+      console.log(`${this.name} has begun sprint challenge on ${subject}.`);
+    }
+  }
+  // #### Project Mananger
+  // * Now that we have instructors and students, we'd be nowhere without our PM's
+  // * ProjectManagers are extensions of Instructors
+  // * ProjectManagers have the following uniqe props:
+  //   * `gradClassName`: i.e. CS1
+  //   * `favInstructor`: i.e. Sean
+  // * ProjectManangers have the following Methods:
+  //   * `standUp` a method that takes in a slack channel and logs `{name} announces to {channel}, @channel standy times!​​​​​
+  //   * `debugsCode` a method that takes in a student object and a subject and logs out `{name} debugs {student.name}'s code on {subject}`
+  class ProjectManager extends Instructor {
+    constructor(props) {
+      super(props);
+      this.gradClassName = props.gradClassName;
+      this.favInstructor = props.favInstructor;
+    }
+  
+    standUp(channel) {
+      console.log(`${this.name} announces to ${channel}, @channel standy times!`);
+    }
+    debugsCode(student, subject) {
+      console.log(`${this.name} debugs ${student.name}'s code on ${subject}.`);
+    }
+  }
```

Comment:
Syntax is perfect, and your properly extending the correct classes. Very well done here! :100: